### ISSUE TITLE
ChoFixedLengthRecordWriter issue with handling double quotes

### ISSUE THREAD

**Issue by kr-prash** [Cinchoo/ChoETL#155]
We are facing two issues with ChoFixedLengthRecordWriter, as detailed below, if the field value contains a double quote.

**ISSUE #1:** 
When the field value contains a double quote character, the ChoFixedLengthRecordWriter is doubling the double quote character by default. Since it adds additional double quote characters, it increases the size of that column where the double quote is present and shifts the starting positions of the subsequent columns. Since it is a fixed length file, it should use the actual size specified for that column. 

For example, if below is the model data,

```
List<Employee> employee = new List<Employee>()
{
	new Employee() { Id = 20, Name = "John Smith", Address = "PO BOX 12165", Age = "25" },
	new Employee() { Id = 21, Name = "Bob Kevin", Address = "123 NEW LIVERPOOL RD \"APT 12\"", Age = "30" },
	new Employee() { Id = 22, Name = "Jack Robert", Address = "PO BOX 123", Age = "40" }
};
```

It is generating the fixed length file with below contents. Here, the Age column value for the second record is shifted by two positions since it added two double quotes (one for each occurance of doule quote in the Address column of the second record).

![image](https://user-images.githubusercontent.com/47215805/135119711-15c337f8-e203-43fe-9652-cc20b27eae63.png)

**ISSUE #2:**
We basically don't want to double the double quote if it is present in the field value. It would be great if you can add a configuration setting to enable or disable this feature. **We would like to have this configuration for both CSV and Fixed Length Writer.** 

In order to handle this issue temporarily, we are basically setting the 'QuoteChar' configuration to a value that is not expected in the flat file, as shown below. Since '~' (tilde) character never occurs in the file, it is able to bypass the below if condition that is replacing the double quote with two double quotes. 

// Temporary Fix
config.QuoteChar = '~';  // tilde character

```
// ChoFixedLengthRecordWriter implementation
if (fieldValue.Contains(Configuration.QuoteChar))
{
	fieldValue = fieldValue.Replace(Configuration.QuoteChar.ToString(), Configuration.DoubleQuoteChar);
}
```

But since we are changing the QuoteChar configuration to tilde character, the ChoFixedLengthRecordWriter will end up using that character to enclose the field value (it adds '~' character to the starting and ending position of the field value) if the "QuoteField" column configuration is set to 'true', which is not correct. So having a configuration setting to enable or disable the feature to replace single double quote with two double quotes would solve this problem. Please help.

```
// ChoFixedLengthRecordWriter implementation
if (quoteValue)
fieldValue = "{1}{0}{1}".FormatString(fieldValue, Configuration.QuoteChar);
```

**Comment by Cinchoo** [comment]
Applied fix, released 1.2.1.26-alpha to nuget. try and let me know.

There is new configuration property added `EscapeQuoteAndDelimiter` to skip the escape operation on quotes.

**Comment by Cinchoo** [comment]
Here is working sample

```
List<EmpRec> employees = new List<EmpRec>()
{
new EmpRec() { Id = 20, Name = "John Smith",  Address = "PO BOX 12165", Age = "25" },
new EmpRec() { Id = 21, Name = "Bob Kevin", Address = "123 NEW LIVERPOOL RD \"APT 12\"", Age = "30" },
new EmpRec() { Id = 22, Name = "Jack Robert", Address = "PO BOX 123", Age = "40" }
};
using (var w = new ChoFixedLengthWriter<EmpRec>(Console.Out)
    .WithFirstLineHeader()
    .Configure(c => c.EscapeQuoteAndDelimiter = false)
    .WithField(f => f.Id, 1, 10)
    .WithField(f => f.Name, 11, 25)
    .WithField(f => f.Address, 36, 50)
    .WithField(f => f.Age, 86, 10)
    )
{
    w.Write(employees);
}
```

Output:

![image](https://user-images.githubusercontent.com/9448455/135174877-eec9ff0a-b9c8-4881-aef3-70a31d41b140.png)

**Comment by kr-prash** [comment]
@Cinchoo, 
Thank you so much for providing a quick fix. We are able to use `EscapeQuoteAndDelimiter` configuration to skip the escape operation on double quotes. Both the scenarios are now working fine, but we are currently doing some testing for other possible scenarios. Will provide you an update by tomorrow. Thank you so much again! :)

**Comment by kr-prash** [comment]
@Cinchoo,
I would like to confirm that the fixes are working fine for us. Thank you so much again for providing the fix so quickly!

**Comment by kr-prash** [comment]
Hi Cinchoo,
Would you please let us know when will the production build with 1.2.1.26-alpha changes be available?

**Comment by Cinchoo** [comment]
Done, released v1.2.1.26 today

**Comment by kr-prash** [comment]
Hi Cinchoo,
Thank you so much for providing the build!
